### PR TITLE
Dispose underlying stream in TarReader.DisposeAsync() as well

### PR DESCRIPTION
**Pull request by akoeplinger**
Same as #79899 but for the async dispose method.

In the System.Formats.Tar WriteEntry_LongFileSize/WriteEntry_LongFileSizeAsync tests the stream passed to `WrappedStream` was also not disposed, causing it to stick around longer (see https://github.com/dotnet/runtime/pull/79907). While looking at adding that I found out we have two nearly identical copies of the helper class in both System.Formats.Tar and System.IO.Compression, but a test in the latter one relies on `WrappedStream` not disposing the stream internally so I just consolidated to a single source file and and separately dispose of the stream passed to `WrappedStream` in the System.Formats.Tar tests.

### CODE REVIEW

File: src/libraries/Common/tests/System/IO/WrappedStream.cs [hunk 1]
```diff
@@ -1,9 +1,7 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.IO;
-
-namespace System.Formats.Tar
+namespace System.IO
 {
     public class WrappedStream : Stream
```
Comment (akoeplinger) [inline src/libraries/Common/tests/System/IO/WrappedStream.cs:6]: @carlossanlop do you remember which changes you meant back in https://github.com/dotnet/runtime/pull/67883#discussion_r847924992

It seems to work with the consolidated version.

**Review by adamsitnik — APPROVED** [review]
LGTM, big thanks for your help and the fix @akoeplinger !

**Comment by adamsitnik** [comment]
Some of the outerloops tests have failed due to timeouts:

```log
  Discovering: System.Formats.Tar.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Formats.Tar.Tests (found 2 of 635 test cases)
  Starting:    System.Formats.Tar.Tests (parallel test collections = on, max threads = 2)
   System.Formats.Tar.Tests: [Long Running Test] 'System.Formats.Tar.Tests.TarWriter_WriteEntry_LongFile_Tests.WriteEntry_LongFileSize', Elapsed: 00:03:59
   System.Formats.Tar.Tests: [Long Running Test] 'System.Formats.Tar.Tests.TarWriter_WriteEntry_LongFile_Tests.WriteEntry_LongFileSize', Elapsed: 00:06:00
   System.Formats.Tar.Tests: [Long Running Test] 'System.Formats.Tar.Tests.TarWriter_WriteEntry_LongFile_Tests.WriteEntry_LongFileSize', Elapsed: 00:08:00
   System.Formats.Tar.Tests: [Long Running Test] 'System.Formats.Tar.Tests.TarWriter_WriteEntry_LongFile_Tests.WriteEntry_LongFileSize', Elapsed: 00:10:00
   System.Formats.Tar.Tests: [Long Running Test] 'System.Formats.Tar.Tests.TarWriter_WriteEntryAsync_LongFile_Tests.WriteEntry_LongFileSizeAsync', Elapsed: 00:02:08
   System.Formats.Tar.Tests: [Long Running Test] 'System.Formats.Tar.Tests.TarWriter_WriteEntryAsync_LongFile_Tests.WriteEntry_LongFileSizeAsync', Elapsed: 00:04:08
['System.Formats.Tar.Tests' END OF WORK ITEM LOG: Command timed out, and was killed]
```

I am going to merge this PR as it's fixing a product bug and reopen #77012 to keep track of the required test changes.

**Pull request merged**